File: android/app/src/main/kotlin/com/transport_system/ts_admin/data_providers/models/oto/ConversationModel.kt
package com.transport_system.ts_admin.data_providers.models.oto

import com.google.gson.annotations.SerializedName
import com.transport_system.ts_admin.data_providers.models.common.ParticipantModel

data class ConversationModel(
    val id: Int?,
    @SerializedName("last_messaged_at") val lastMessagedAt: Int?,
    @SerializedName("receiver") val user: ConversationReceiverModel?,
    @SerializedName("date_time_in_humans") val dateTimeInHumans: String?,
    val name: String?,
    @SerializedName("chat_able") val chatAble: Boolean?,
    val status: String?,
    val participants: List<ParticipantModel>?,
    @SerializedName("notification_muted") val notificationMuted: Boolean?
)

File: ios/Runner/Database/Models/Oto/ConversationModel.swift
//
//  ConversationModel.swift
//  Runner
//
//  Created by Hashim Khan on 16/12/2024.
//


import Foundation

struct ConversationModel: Codable, Equatable {
    var id: Int?
    var lastMessagedAt: Int?
    var user: ConversationReciverModel?
    var dateTimeInHumans: String?
    var name: String?
    var chatAble: Bool?
    var status: String?
    var participants: [ParticipantModel]?
    var notificationMuted: Bool?

    enum CodingKeys: String, CodingKey {
        case id
        case lastMessagedAt = "last_messaged_at"
        case user = "receiver"
        case dateTimeInHumans = "date_time_in_humans"
        case name
        case chatAble = "chat_able"
        case status
        case participants
        case notificationMuted = "notification_muted"
    }
}



